src/components/map/components/bonus-program/BonusProgramMapButton.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import {Button} from '@atb/components/button';
+import {StyleSheet, useThemeContext} from '@atb/theme';
+import {useAnalyticsContext} from '@atb/analytics';
+import {FavoriteFill} from '@atb/assets/svg/mono-icons/places';
+
+type BonusProgramMapButtonProps = {
+  onPress: () => void;
+  isLoading: boolean;
+  bonusPoints: number;
+};
+export const BonusProgramMapButton = ({
+  onPress,
+  isLoading,
+  bonusPoints,
```

Comment:
It is hard to know how the future will look, but I think plan A can be to keep as much internally in the component as possible. So if the button itself are able to retrieve the `bonusPoints` through useQuery or React.Context then the parent won't need to hassle with retrieving it.